### PR TITLE
Check bash version for hack/e2e.sh

### LINKED ISSUE #1560: e2e test doesn't work with default version of bash on macOS
## Feature Request

**Is your feature request related to a problem? Please describe:**

When run `./hack/e2e.sh` with the default bash on macOS (version 3.2), there's a error message:
```bash
./hack/e2e.sh: line 136: declare: -A: invalid option
```

It is because bash v3.2 doesn't support `declare  -A`. 

Either the developer should upgrade their bash or some shell script which compatible with bash v3.2 should be used here.

**Describe the feature you'd like:**

Use a method which also works in bash v3.2 or describe this in doc.

### PR DESCRIPTION
**Pull request by longfangsong**
Add bash version requirements note in doc.
fix #1560

### What problem does this PR solve? 
Check bash version in e2e test script.
It add bash version requirements note in doc.
### What is changed and how does it work?

`e2e.sh` and `CONTRIBUTIN.md`.

In `e2e.sh` , we check the `$BASH_VERSION` and make sure it >= 4 before really execute anything.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -26,6 +26,16 @@ cd $ROOT
 
 source "${ROOT}/hack/lib.sh"
 
+# check bash version
+BASH_MAJOR_VERSION=$(echo "$BASH_VERSION" | cut -d '.' -f 1)
+# we need bash version >= 4
```
Comment (Yisaer) [inline hack/e2e.sh]: Could we merge these 2 notes into on note?

Comment (longfangsong) [reply hack/e2e.sh]: The first one explain the usage of the following "block" and the second one explain the usage of the following "statement".

But it is not that obvious so I think it is ok to merge.

**Comment by cofyc** [comment]
https://github.com/pingcap/tidb-operator/pull/1562#discussion_r367204433

**Comment by cofyc** [comment]
Thanks for the fix! This PR can guide people in local development.
If you think it's better to allow `./hack/e2e.sh -h` to run or want to improve comments, etc., feel free to submit another PR.

**Pull request merged**